Index: libs/picomodel/pm_ase.c
===================================================================
--- libs/picomodel/pm_ase.c	(revision 3696)
+++ libs/picomodel/pm_ase.c	(working copy)
@@ -1131,27 +1131,31 @@
           {
             if(_pico_strnicmp(p, "quake", 5) == 0 || _pico_strnicmp(p, "doom", 4) == 0)
             {
-              break;
+              /* root-relative */
+              for(; *p != '\0'; ++p)
+              {
+                if(*p == '/')
+                {
+                  ++p;
+                  /* game-relative */
+                  for(; *p != '\0'; ++p)
+                  {
+                    if(*p == '/')
+                    {
+                      ++p;
+                      break;
+                    }
+                  }
+                }
+              }
             }
-          }
-          /* root-relative */
-          for(; *p != '\0'; ++p)
-          {
-            if(*p == '/')
+            /* DoomEdit's ASE loader searches for /base/ */
+            else if(_pico_strnicmp(p, "base/", 5) == 0)
             {
-              ++p;
+              p += 5;
               break;
             }
           }
-          /* game-relative */
-          for(; *p != '\0'; ++p)
-          {
-            if(*p == '/')
-            {
-              ++p;
-              break;
-            }
-          }
 
           if(*p != '\0')
           {
Index: plugins/archivezip/plugin.cpp
===================================================================
--- plugins/archivezip/plugin.cpp	(revision 3696)
+++ plugins/archivezip/plugin.cpp	(working copy)
@@ -29,6 +29,38 @@
 
 #include "ZipArchive.h"
 
+class ArchivePK3API :
+	public ArchiveLoader
+{
+public:
+	// greebo: Returns the opened file or NULL if failed.
+	virtual ArchivePtr openArchive(const std::string& name) {
+		return ZipArchivePtr(new ZipArchive(name));
+	}
+		
+	virtual const std::string& getExtension() {
+		static std::string _ext("PK3");
+		return _ext;
+	}
+  
+	// RegisterableModule implementation
+	virtual const std::string& getName() const {
+		static std::string _name("ArchivePK3");
+		return _name;
+	}
+
+	virtual const StringSet& getDependencies() const {
+		static StringSet _dependencies; // no dependencies
+		return _dependencies;
+	}
+
+	virtual void initialiseModule(const ApplicationContext& ctx) {
+		globalOutputStream() << "ArchivePK3::initialiseModule called\n";
+	}
+};
+typedef boost::shared_ptr<ArchivePK3API> ArchivePK3APIPtr;
+
+
 class ArchivePK4API :
 	public ArchiveLoader
 {
@@ -61,6 +93,7 @@
 typedef boost::shared_ptr<ArchivePK4API> ArchivePK4APIPtr;
 
 extern "C" void DARKRADIANT_DLLEXPORT RegisterModule(IModuleRegistry& registry) {
+	registry.registerModule(ArchivePK3APIPtr(new ArchivePK3API));
 	registry.registerModule(ArchivePK4APIPtr(new ArchivePK4API));
 	
 	// Initialise the streams
Index: plugins/image/image.cpp
===================================================================
--- plugins/image/image.cpp	(revision 3696)
+++ plugins/image/image.cpp	(working copy)
@@ -26,6 +26,7 @@
 
 #include "jpeg.h"
 #include "tga.h"
+#include "png.h"
 #include "bmp.h"
 #include "pcx.h"
 #include "dds.h"
@@ -33,6 +34,7 @@
 
 typedef boost::shared_ptr<TGALoader> TGALoaderPtr;
 typedef boost::shared_ptr<JPGLoader> JPGLoaderPtr;
+typedef boost::shared_ptr<PNGLoader> PNGLoaderPtr;
 typedef boost::shared_ptr<PCXLoader> PCXLoaderPtr;
 typedef boost::shared_ptr<BMPLoader> BMPLoaderPtr;
 typedef boost::shared_ptr<DDSLoader> DDSLoaderPtr;
@@ -41,6 +43,7 @@
 extern "C" void DARKRADIANT_DLLEXPORT RegisterModule(IModuleRegistry& registry) {
 	registry.registerModule(TGALoaderPtr(new TGALoader));
 	registry.registerModule(JPGLoaderPtr(new JPGLoader));
+	registry.registerModule(PNGLoaderPtr(new PNGLoader));
 	registry.registerModule(PCXLoaderPtr(new PCXLoader));
 	registry.registerModule(BMPLoaderPtr(new BMPLoader));
 	registry.registerModule(DDSLoaderPtr(new DDSLoader));
Index: plugins/image/Makefile.am
===================================================================
--- plugins/image/Makefile.am	(revision 3696)
+++ plugins/image/Makefile.am	(working copy)
@@ -12,5 +12,6 @@
                    jpeg.cpp \
                    ImageGDK.cpp \
                    pcx.cpp \
+                   png.cpp \
                    tga.cpp
 
Index: plugins/image/png.cpp
===================================================================
--- plugins/image/png.cpp	(revision 0)
+++ plugins/image/png.cpp	(revision 0)
@@ -0,0 +1,169 @@
+/*
+Copyright (C) 1999-2006 Id Software, Inc. and contributors.
+For a list of contributors, see the accompanying CONTRIBUTORS file.
+
+This file is part of GtkRadiant.
+
+GtkRadiant is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+GtkRadiant is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GtkRadiant; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include "png.h"
+
+#include <png.h>
+//#include <setjmp.h>
+//#include <stdlib.h>
+//#include <stdio.h>
+//#include <string.h>
+#include "stream/textstream.h"
+
+#include "ifilesystem.h"
+
+#include "imagelib.h"
+
+typedef unsigned char byte;
+
+void user_warning_fn(png_structp png_ptr, png_const_charp warning_msg)
+{
+  globalErrorStream() << "libpng warning: " << warning_msg << "\n";
+}
+
+void user_error_fn(png_structp png_ptr, png_const_charp error_msg)
+{
+  globalErrorStream() << "libpng error: " << error_msg << "\n";
+  longjmp(png_ptr->jmpbuf, 0);
+}
+
+void user_read_data(png_structp png_ptr, png_bytep data, png_uint_32 length)
+{
+  png_bytep *p_p_fbuffer = (png_bytep*)png_get_io_ptr(png_ptr);
+  memcpy(data, *p_p_fbuffer, length);
+  *p_p_fbuffer += length;
+}
+
+static RGBAImagePtr LoadPNGBuff (unsigned char* fbuffer)
+{
+  png_byte** row_pointers;
+  png_bytep p_fbuffer;
+
+  p_fbuffer = fbuffer;
+
+  // the reading glue
+  // http://www.libpng.org/pub/png/libpng-manual.html
+
+  png_structp png_ptr = png_create_read_struct
+    (PNG_LIBPNG_VER_STRING, (png_voidp)NULL,
+    user_error_fn, user_warning_fn);
+  if (!png_ptr)
+  {
+    globalErrorStream() << "libpng error: png_create_read_struct\n";
+    RGBAImagePtr();
+  }
+
+  png_infop info_ptr = png_create_info_struct(png_ptr);
+  if (!info_ptr) {
+    png_destroy_read_struct(&png_ptr,
+      (png_infopp)NULL, (png_infopp)NULL);
+    globalErrorStream() << "libpng error: png_create_info_struct (info_ptr)\n";
+    RGBAImagePtr();
+  }
+
+  png_infop end_info = png_create_info_struct(png_ptr);
+  if (!end_info) {
+    png_destroy_read_struct(&png_ptr, &info_ptr,
+      (png_infopp)NULL);
+    globalErrorStream() << "libpng error: png_create_info_struct (end_info)\n";
+    RGBAImagePtr();
+  }
+
+  // configure the read function
+  png_set_read_fn(png_ptr, (voidp)&p_fbuffer, (png_rw_ptr)&user_read_data);
+
+  if (setjmp(png_ptr->jmpbuf)) {
+    png_destroy_read_struct(&png_ptr, &info_ptr,
+      &end_info);
+    RGBAImagePtr();
+  }
+
+  png_read_info(png_ptr, info_ptr);
+
+  int bit_depth = png_get_bit_depth(png_ptr, info_ptr);
+  int color_type = png_get_color_type(png_ptr, info_ptr);
+
+  // we want to treat all images the same way
+  //   The following code transforms grayscale images of less than 8 to 8 bits,
+  //   changes paletted images to RGB, and adds a full alpha channel if there is
+  //   transparency information in a tRNS chunk.
+  if (color_type == PNG_COLOR_TYPE_PALETTE)
+   png_set_palette_to_rgb(png_ptr);
+
+  if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8)
+    png_set_gray_1_2_4_to_8(png_ptr);
+
+  if (png_get_valid(png_ptr, info_ptr, PNG_INFO_tRNS))
+    png_set_tRNS_to_alpha(png_ptr);
+
+  if ( ! ( color_type & PNG_COLOR_MASK_ALPHA ) ) {
+    // Set the background color to draw transparent and alpha images over.
+    png_color_16 my_background, *image_background;
+
+    if (png_get_bKGD(png_ptr, info_ptr, &image_background))
+      png_set_background(png_ptr, image_background,
+      PNG_BACKGROUND_GAMMA_FILE, 1, 1.0);
+    else
+      png_set_background(png_ptr, &my_background,
+      PNG_BACKGROUND_GAMMA_SCREEN, 0, 1.0);
+
+    // Add alpha byte after each RGB triplet
+    png_set_filler(png_ptr, 0xff, PNG_FILLER_AFTER);
+  }
+
+  // read the sucker in one chunk
+  png_read_update_info(png_ptr, info_ptr);
+
+  color_type = png_get_color_type(png_ptr, info_ptr);
+  bit_depth = png_get_bit_depth(png_ptr, info_ptr);
+
+  int width = png_get_image_width(png_ptr, info_ptr);
+  int height = png_get_image_height(png_ptr, info_ptr);
+
+  // allocate the pixel buffer, and the row pointers
+  RGBAImagePtr image(new RGBAImage(width, height));
+
+  row_pointers = (png_byte**) malloc((height) * sizeof(png_byte*));
+
+  int i;
+  for(i = 0; i < (height); i++)
+    row_pointers[i] = (png_byte*)(image->getRGBAPixels()) + i * 4 * (width);
+
+  // actual read
+  png_read_image(png_ptr, row_pointers);
+
+  /* read rest of file, and get additional chunks in info_ptr - REQUIRED */
+  png_read_end(png_ptr, info_ptr);
+
+  /* free up the memory structure */
+  png_destroy_read_struct(&png_ptr, &info_ptr, png_infopp_NULL);
+
+  free(row_pointers);
+
+  return image;
+}
+
+ImagePtr LoadPNG(ArchiveFile& file)
+{
+  ScopedArchiveBuffer buffer(file);
+  return LoadPNGBuff(buffer.buffer);//, static_cast<int>(buffer.length));
+}
+
Index: plugins/image/png.h
===================================================================
--- plugins/image/png.h	(revision 0)
+++ plugins/image/png.h	(revision 0)
@@ -0,0 +1,77 @@
+/*
+Copyright (C) 1999-2006 Id Software, Inc. and contributors.
+For a list of contributors, see the accompanying CONTRIBUTORS file.
+
+This file is part of GtkRadiant.
+
+GtkRadiant is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+GtkRadiant is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GtkRadiant; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#if !defined (INCLUDED_PNG_H)
+#define INCLUDED_PNG_H
+
+#include "ifilesystem.h"
+#include "iimage.h"
+#include "imagelib.h" // for RGBAImagePtr
+#include <iostream>
+
+ImagePtr LoadPNG(ArchiveFile& file);
+
+/* Tr3B: A PNGLoader is capable of loading Portable Network Graphic (PNG) files.
+ *  
+ * Use load() to actually retrieve an Image* object with the loaded image.
+ * 
+ * Shouldn't be used to load textures directly, use the 
+ * GlobalShaderSystem() module instead.  
+ * 
+ * Complies with the ImageLoader interface defined in "iimage.h" 
+ */
+class PNGLoader :
+	public ImageLoader
+{
+public:
+	/* greebo: This loads the file and returns the pointer to 
+	 * the allocated Image object (or NULL, if the load failed). 
+	 */
+	ImagePtr load(ArchiveFile& file) const {
+		// Pass the call to the according load function
+		return LoadPNG(file);
+	}
+	
+	/* greebo: Gets the file extension of the supported image file type (e.g. "jpg") 
+	 */
+	std::string getExtension() const {
+		return "png";
+	}
+
+	// RegisterableModule implementation
+	virtual const std::string& getName() const {
+		static std::string _name("ImageLoaderPNG");
+		return _name;
+	}
+	
+	virtual const StringSet& getDependencies() const {
+		static StringSet _dependencies; // no dependencies
+  		return _dependencies;
+	}
+	
+	virtual void initialiseModule(const ApplicationContext& ctx) {
+		globalOutputStream() << "ImageLoaderPNG::initialiseModule called.\n";
+	}
+};
+
+#endif
+
+
Index: plugins/vfspk3/Doom3FileSystem.cpp
===================================================================
--- plugins/vfspk3/Doom3FileSystem.cpp	(revision 3696)
+++ plugins/vfspk3/Doom3FileSystem.cpp	(working copy)
@@ -110,7 +110,7 @@
 	globalOutputStream() << "[vfs] searched directory: " << path.c_str() << "\n";
 	
 	// Get the ArchiveLoader and try to load each file
-	ArchiveLoader& archiveModule = GlobalArchive("PK4");  
+	ArchiveLoader& archiveModule = GlobalArchive("PK3");  
 	
 	// add the entries to the vfs
 	for (SortedFilenames::iterator i = filenameList.begin(); i != filenameList.end(); ++i) {
@@ -285,7 +285,7 @@
 	static StringSet _dependencies;
 
 	if (_dependencies.empty()) {
-		_dependencies.insert("ArchivePK4");
+		_dependencies.insert("ArchivePK3");
 		_dependencies.insert(MODULE_GAMEMANAGER);
 	}
 
Index: SConscript
===================================================================
--- SConscript	(revision 3696)
+++ SConscript	(working copy)
@@ -210,7 +210,7 @@
 skinsEnv.Install(INSTALL + '/modules', skinsLib)
 
 image_env = module_env.Copy()
-image_lst = build_list('plugins/image', 'bmp.cpp jpeg.cpp image.cpp pcx.cpp tga.cpp dds.cpp ImageGDK.cpp')
+image_lst = build_list('plugins/image', 'bmp.cpp jpeg.cpp image.cpp pcx.cpp tga.cpp dds.cpp ImageGDK.cpp png.cpp')
 image_env.useGtk2()
 image_env.useGlib2()
 image_env.Append(LIBS=['jpeg6', 'ddslib'])
Index: tools/vcprojects/archivezip.vcproj
===================================================================
--- tools/vcprojects/archivezip.vcproj	(revision 3696)
+++ tools/vcprojects/archivezip.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="archivezip"
 	ProjectGUID="{A7E0FE03-E9BB-4478-9752-250BBD406C2D}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/DarkRadiant.sln
===================================================================
--- tools/vcprojects/DarkRadiant.sln	(revision 3696)
+++ tools/vcprojects/DarkRadiant.sln	(working copy)
@@ -1,16 +1,16 @@
-Microsoft Visual Studio Solution File, Format Version 9.00
-# Visual Studio 2005
+Microsoft Visual Studio Solution File, Format Version 10.00
+# Visual C++ Express 2008
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "mathlib", "mathlib.vcproj", "{3C9FB5AA-7118-476E-B33D-D3AC1C8412BB}"
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "DarkRadiant", "DarkRadiant.vcproj", "{8E70385C-223A-4DD1-9B99-28FF2331A2B5}"
 	ProjectSection(ProjectDependencies) = postProject
 		{4286E605-5358-462F-AC57-44FCC697585B} = {4286E605-5358-462F-AC57-44FCC697585B}
 		{F7408B46-E4A9-470C-9731-9A1564247385} = {F7408B46-E4A9-470C-9731-9A1564247385}
-		{A15EFB56-927F-411D-A57B-0328321456A2} = {A15EFB56-927F-411D-A57B-0328321456A2}
-		{B6D4B38A-0C39-42CD-8193-75979E1F4D68} = {B6D4B38A-0C39-42CD-8193-75979E1F4D68}
-		{3C9FB5AA-7118-476E-B33D-D3AC1C8412BB} = {3C9FB5AA-7118-476E-B33D-D3AC1C8412BB}
-		{8050CFB1-20CC-40FD-8D74-0901C9CA2BE8} = {8050CFB1-20CC-40FD-8D74-0901C9CA2BE8}
 		{D42098ED-E192-4DC1-BA1C-80812AB194B4} = {D42098ED-E192-4DC1-BA1C-80812AB194B4}
+		{8050CFB1-20CC-40FD-8D74-0901C9CA2BE8} = {8050CFB1-20CC-40FD-8D74-0901C9CA2BE8}
+		{3C9FB5AA-7118-476E-B33D-D3AC1C8412BB} = {3C9FB5AA-7118-476E-B33D-D3AC1C8412BB}
+		{B6D4B38A-0C39-42CD-8193-75979E1F4D68} = {B6D4B38A-0C39-42CD-8193-75979E1F4D68}
+		{A15EFB56-927F-411D-A57B-0328321456A2} = {A15EFB56-927F-411D-A57B-0328321456A2}
 	EndProjectSection
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "archivezip", "archivezip.vcproj", "{A7E0FE03-E9BB-4478-9752-250BBD406C2D}"
@@ -71,8 +71,8 @@
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "sound", "sound.vcproj", "{1482AAB4-40B6-44C0-9FD8-AF80C18A5E7F}"
 	ProjectSection(ProjectDependencies) = postProject
-		{A15EFB56-927F-411D-A57B-0328321456A2} = {A15EFB56-927F-411D-A57B-0328321456A2}
 		{B6D4B38A-0C39-42CD-8193-75979E1F4D68} = {B6D4B38A-0C39-42CD-8193-75979E1F4D68}
+		{A15EFB56-927F-411D-A57B-0328321456A2} = {A15EFB56-927F-411D-A57B-0328321456A2}
 	EndProjectSection
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "uimanager", "uimanager.vcproj", "{97E454CC-E8B0-4274-AEAE-5157F420C984}"
@@ -263,9 +263,7 @@
 		{3BBB3A48-F1A9-4660-93C1-D141CB07E2DE}.Release|Win32.ActiveCfg = Release|Win32
 		{3BBB3A48-F1A9-4660-93C1-D141CB07E2DE}.Release|Win32.Build.0 = Release|Win32
 		{41CAAD78-9E48-4C6E-B8F8-A04A859C16F9}.Debug|Win32.ActiveCfg = Debug|Win32
-		{41CAAD78-9E48-4C6E-B8F8-A04A859C16F9}.Debug|Win32.Build.0 = Debug|Win32
 		{41CAAD78-9E48-4C6E-B8F8-A04A859C16F9}.Release|Win32.ActiveCfg = Release|Win32
-		{41CAAD78-9E48-4C6E-B8F8-A04A859C16F9}.Release|Win32.Build.0 = Release|Win32
 	EndGlobalSection
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
Index: tools/vcprojects/DarkRadiant.vcproj
===================================================================
--- tools/vcprojects/DarkRadiant.vcproj	(revision 3696)
+++ tools/vcprojects/DarkRadiant.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="DarkRadiant"
 	ProjectGUID="{8E70385C-223A-4DD1-9B99-28FF2331A2B5}"
 	RootNamespace="GtkRadiant"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -86,8 +87,10 @@
 				GenerateMapFile="false"
 				SubSystem="1"
 				EntryPointSymbol="mainCRTStartup"
-				TargetMachine="1"
+				RandomizedBaseAddress="1"
 				FixedBaseAddress="1"
+				DataExecutionPrevention="0"
+				TargetMachine="1"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -108,9 +111,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine="$(SolutionDir)/post_build_event.bat"
 			/>
@@ -185,8 +185,10 @@
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
 				EntryPointSymbol="mainCRTStartup"
-				TargetMachine="1"
+				RandomizedBaseAddress="1"
 				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
+				TargetMachine="1"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -207,9 +209,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine="$(SolutionDir)/post_build_event.bat"
 			/>
Index: tools/vcprojects/ddslib.vcproj
===================================================================
--- tools/vcprojects/ddslib.vcproj	(revision 3696)
+++ tools/vcprojects/ddslib.vcproj	(working copy)
@@ -1,10 +1,11 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="ddslib"
 	ProjectGUID="{D42098ED-E192-4DC1-BA1C-80812AB194B4}"
 	RootNamespace="jpeglib"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
Index: tools/vcprojects/dm.d3hook.vcproj
===================================================================
--- tools/vcprojects/dm.d3hook.vcproj	(revision 3696)
+++ tools/vcprojects/dm.d3hook.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="dm.d3hook"
 	ProjectGUID="{916F8AE5-D7FC-44E1-908D-B54100C6F4AD}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/dm.difficulty.vcproj
===================================================================
--- tools/vcprojects/dm.difficulty.vcproj	(revision 3696)
+++ tools/vcprojects/dm.difficulty.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="dm.difficulty"
 	ProjectGUID="{68B95079-09DF-48F7-BAD9-81C33118BEFD}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/dm.objectives.vcproj
===================================================================
--- tools/vcprojects/dm.objectives.vcproj	(revision 3696)
+++ tools/vcprojects/dm.objectives.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="dm.objectives"
 	ProjectGUID="{05E9A613-2108-4843-A567-16B432DAC272}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/dm.stimresponse.vcproj
===================================================================
--- tools/vcprojects/dm.stimresponse.vcproj	(revision 3696)
+++ tools/vcprojects/dm.stimresponse.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="dm.stimresponse"
 	ProjectGUID="{EE9A3B61-58E0-4AA4-8942-6CFCBFAACC3F}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/eclassmgr.vcproj
===================================================================
--- tools/vcprojects/eclassmgr.vcproj	(revision 3696)
+++ tools/vcprojects/eclassmgr.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="eclassmgr"
 	ProjectGUID="{4CF3ED90-BE1A-45D1-9DC7-7CC6884A9754}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/eclasstree.vcproj
===================================================================
--- tools/vcprojects/eclasstree.vcproj	(revision 3696)
+++ tools/vcprojects/eclasstree.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="eclasstree"
 	ProjectGUID="{3B1A6CA3-D76E-4020-860D-1BED3DF7A3BE}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/entity.vcproj
===================================================================
--- tools/vcprojects/entity.vcproj	(revision 3696)
+++ tools/vcprojects/entity.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="entity"
 	ProjectGUID="{DC28A0B7-51AB-4F4A-A428-0765CA2CE6AB}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/entitylist.vcproj
===================================================================
--- tools/vcprojects/entitylist.vcproj	(revision 3696)
+++ tools/vcprojects/entitylist.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="entitylist"
 	ProjectGUID="{A35C6082-EE2D-4DEB-944B-61CFEB19C5B1}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/eventmanager.vcproj
===================================================================
--- tools/vcprojects/eventmanager.vcproj	(revision 3696)
+++ tools/vcprojects/eventmanager.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="eventmanager"
 	ProjectGUID="{0EB2B264-13C5-47E8-B99A-A46C8C2E108F}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/filetypes.vcproj
===================================================================
--- tools/vcprojects/filetypes.vcproj	(revision 3696)
+++ tools/vcprojects/filetypes.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="filetypes"
 	ProjectGUID="{A0E26F34-9447-4B04-9069-A6517F065759}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -81,6 +82,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -103,9 +106,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -177,9 +177,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -200,9 +202,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/filters.vcproj
===================================================================
--- tools/vcprojects/filters.vcproj	(revision 3696)
+++ tools/vcprojects/filters.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="filters"
 	ProjectGUID="{52681EA4-8879-4542-A394-4CFDD17FE275}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/grid.vcproj
===================================================================
--- tools/vcprojects/grid.vcproj	(revision 3696)
+++ tools/vcprojects/grid.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="grid"
 	ProjectGUID="{859C6FA8-FA3A-47D4-A64C-9B3D4282FCA8}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -81,6 +82,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -103,9 +106,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -178,9 +178,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -201,9 +203,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/gtkutillib.vcproj
===================================================================
--- tools/vcprojects/gtkutillib.vcproj	(revision 3696)
+++ tools/vcprojects/gtkutillib.vcproj	(working copy)
@@ -1,10 +1,11 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="gtkutillib"
 	ProjectGUID="{B6D4B38A-0C39-42CD-8193-75979E1F4D68}"
 	RootNamespace="gtkutillib"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
Index: tools/vcprojects/image.vcproj
===================================================================
--- tools/vcprojects/image.vcproj	(revision 3696)
+++ tools/vcprojects/image.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="image"
 	ProjectGUID="{002AEFE8-2CA3-43CA-84CA-CF74FF9F1E8E}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -44,7 +45,7 @@
 				Name="VCCLCompilerTool"
 				AdditionalOptions="/EHs"
 				Optimization="0"
-				AdditionalIncludeDirectories="&quot;$(SolutionDir)/../../include&quot;;&quot;$(SolutionDir)/../../libs&quot;;&quot;$(SolutionDir)/../../w32deps/boost/include&quot;;&quot;$(SolutionDir)/../../w32deps/libxml2/include&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/gtk-2.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/atk-1.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/glib-2.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/cairo&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/pango-1.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/lib/glib-2.0/include&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/lib/gtk-2.0/include&quot;"
+				AdditionalIncludeDirectories="&quot;$(SolutionDir)/../../include&quot;;&quot;$(SolutionDir)/../../libs&quot;;&quot;$(SolutionDir)/../../w32deps/boost/include&quot;;&quot;$(SolutionDir)/../../w32deps/libxml2/include&quot;;&quot;$(SolutionDir)/../../w32deps/libpng/include&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/gtk-2.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/atk-1.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/glib-2.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/cairo&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/pango-1.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/lib/glib-2.0/include&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/lib/gtk-2.0/include&quot;"
 				PreprocessorDefinitions="_CRT_SECURE_NO_DEPRECATE;WIN32;"
 				StringPooling="true"
 				MinimalRebuild="true"
@@ -73,16 +74,18 @@
 			<Tool
 				Name="VCLinkerTool"
 				AdditionalOptions="/NODEFAULTLIB:LIBCMT"
-				AdditionalDependencies="ddslib.lib jpeglib.lib xmlutillib.lib libxml2.lib glib-2.0.lib gtk-win32-2.0.lib gdk-win32-2.0.lib gobject-2.0.lib gdk_pixbuf-2.0.lib"
+				AdditionalDependencies="ddslib.lib jpeglib.lib xmlutillib.lib libxml2.lib glib-2.0.lib gtk-win32-2.0.lib gdk-win32-2.0.lib gobject-2.0.lib gdk_pixbuf-2.0.lib png.lib"
 				OutputFile="$(OutDir)/$(ProjectName).dll"
 				LinkIncremental="1"
 				SuppressStartupBanner="true"
-				AdditionalLibraryDirectories="&quot;$(SolutionDir)\..\..\build\libs\$(ConfigurationName)&quot;;&quot;$(SolutionDir)\..\..\w32deps/gtk2\lib&quot;;&quot;$(SolutionDir)\..\..\w32deps/libxml2\lib&quot;"
+				AdditionalLibraryDirectories="&quot;$(SolutionDir)\..\..\build\libs\$(ConfigurationName)&quot;;&quot;$(SolutionDir)\..\..\w32deps/gtk2\lib&quot;;&quot;$(SolutionDir)\..\..\w32deps/libxml2\lib&quot;;&quot;$(SolutionDir)\..\..\w32deps/libpng\lib&quot;"
 				IgnoreDefaultLibraryNames=""
 				ModuleDefinitionFile="$(SolutionDir)/../../plugins/$(ProjectName)/$(ProjectName).def"
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -105,9 +108,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -144,7 +144,7 @@
 				InlineFunctionExpansion="2"
 				EnableIntrinsicFunctions="true"
 				FavorSizeOrSpeed="1"
-				AdditionalIncludeDirectories="&quot;$(SolutionDir)/../../include&quot;;&quot;$(SolutionDir)/../../libs&quot;;&quot;$(SolutionDir)/../../w32deps/boost/include&quot;;&quot;$(SolutionDir)/../../w32deps/libxml2/include&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/gtk-2.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/atk-1.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/glib-2.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/cairo&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/pango-1.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/lib/glib-2.0/include&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/lib/gtk-2.0/include&quot;"
+				AdditionalIncludeDirectories="&quot;$(SolutionDir)/../../include&quot;;&quot;$(SolutionDir)/../../libs&quot;;&quot;$(SolutionDir)/../../w32deps/boost/include&quot;;&quot;$(SolutionDir)/../../w32deps/libxml2/include&quot;;&quot;$(SolutionDir)/../../w32deps/libpng/include&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/gtk-2.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/atk-1.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/glib-2.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/cairo&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/include/pango-1.0&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/lib/glib-2.0/include&quot;;&quot;$(SolutionDir)/../../w32deps/gtk2/lib/gtk-2.0/include&quot;"
 				PreprocessorDefinitions="_CRT_SECURE_NO_DEPRECATE;WIN32"
 				StringPooling="true"
 				ExceptionHandling="0"
@@ -168,11 +168,11 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
-				AdditionalDependencies="ddslib.lib jpeglib.lib xmlutillib.lib libxml2.lib glib-2.0.lib gtk-win32-2.0.lib gdk-win32-2.0.lib gobject-2.0.lib gdk_pixbuf-2.0.lib"
+				AdditionalDependencies="ddslib.lib jpeglib.lib xmlutillib.lib libxml2.lib glib-2.0.lib gtk-win32-2.0.lib gdk-win32-2.0.lib gobject-2.0.lib gdk_pixbuf-2.0.lib png.lib"
 				OutputFile="$(OutDir)/$(ProjectName).dll"
 				LinkIncremental="1"
 				SuppressStartupBanner="true"
-				AdditionalLibraryDirectories="&quot;$(SolutionDir)\..\..\build\libs\$(ConfigurationName)&quot;;&quot;$(SolutionDir)\..\..\w32deps/gtk2\lib&quot;;&quot;$(SolutionDir)\..\..\w32deps/libxml2\lib&quot;"
+				AdditionalLibraryDirectories="&quot;$(SolutionDir)\..\..\build\libs\$(ConfigurationName)&quot;;&quot;$(SolutionDir)\..\..\w32deps/gtk2\lib&quot;;&quot;$(SolutionDir)\..\..\w32deps/libxml2\lib&quot;;&quot;$(SolutionDir)\..\..\w32deps/libpng\lib&quot;"
 				IgnoreDefaultLibraryNames=""
 				ModuleDefinitionFile="$(SolutionDir)/../../plugins/$(ProjectName)/$(ProjectName).def"
 				GenerateDebugInformation="true"
@@ -180,9 +180,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -203,9 +205,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -268,6 +267,14 @@
 				>
 			</File>
 			<File
+				RelativePath="..\..\plugins\image\png.cpp"
+				>
+			</File>
+			<File
+				RelativePath="..\..\plugins\image\png.h"
+				>
+			</File>
+			<File
 				RelativePath="..\..\plugins\image\tga.cpp"
 				>
 			</File>
Index: tools/vcprojects/include.vcproj
===================================================================
--- tools/vcprojects/include.vcproj	(revision 3696)
+++ tools/vcprojects/include.vcproj	(working copy)
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="include"
 	ProjectGUID="{41CAAD78-9E48-4C6E-B8F8-A04A859C16F9}"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -48,6 +49,8 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -68,9 +71,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 			/>
 		</Configuration>
@@ -109,6 +109,8 @@
 			/>
 			<Tool
 				Name="VCLinkerTool"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -129,9 +131,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 			/>
 		</Configuration>
Index: tools/vcprojects/jpeglib.vcproj
===================================================================
--- tools/vcprojects/jpeglib.vcproj	(revision 3696)
+++ tools/vcprojects/jpeglib.vcproj	(working copy)
@@ -1,10 +1,11 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="jpeglib"
 	ProjectGUID="{4286E605-5358-462F-AC57-44FCC697585B}"
 	RootNamespace="jpeglib"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
Index: tools/vcprojects/libvorbis/libvorbis.vcproj
===================================================================
--- tools/vcprojects/libvorbis/libvorbis.vcproj	(revision 3696)
+++ tools/vcprojects/libvorbis/libvorbis.vcproj	(working copy)
@@ -1,10 +1,11 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="libvorbis"
 	ProjectGUID="{3BBB3A48-F1A9-4660-93C1-D141CB07E2DE}"
 	RootNamespace="libvorbis"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
Index: tools/vcprojects/libxml2/libxml2.vcproj
===================================================================
--- tools/vcprojects/libxml2/libxml2.vcproj	(revision 3696)
+++ tools/vcprojects/libxml2/libxml2.vcproj	(working copy)
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="libxml2"
 	ProjectGUID="{6780EE95-8768-4138-A89B-47D3504A6E5D}"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
Index: tools/vcprojects/mapdoom3.vcproj
===================================================================
--- tools/vcprojects/mapdoom3.vcproj	(revision 3696)
+++ tools/vcprojects/mapdoom3.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="mapdoom3"
 	ProjectGUID="{3425D3E8-F025-448A-94C8-7EBA6F7A52E5}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/mathlib.vcproj
===================================================================
--- tools/vcprojects/mathlib.vcproj	(revision 3696)
+++ tools/vcprojects/mathlib.vcproj	(working copy)
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="mathlib"
 	ProjectGUID="{3C9FB5AA-7118-476E-B33D-D3AC1C8412BB}"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
Index: tools/vcprojects/md5model.vcproj
===================================================================
--- tools/vcprojects/md5model.vcproj	(revision 3696)
+++ tools/vcprojects/md5model.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="md5model"
 	ProjectGUID="{D39D8067-7DE5-4BC2-8471-CFE99B00B042}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -83,6 +84,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -105,9 +108,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -181,9 +181,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -204,9 +206,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/model.vcproj
===================================================================
--- tools/vcprojects/model.vcproj	(revision 3696)
+++ tools/vcprojects/model.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="model"
 	ProjectGUID="{AFC3BF28-FC48-44C6-A0AA-695865EB4C70}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -83,6 +84,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -105,9 +108,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -181,9 +181,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -204,9 +206,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/particles.vcproj
===================================================================
--- tools/vcprojects/particles.vcproj	(revision 3696)
+++ tools/vcprojects/particles.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="particles"
 	ProjectGUID="{4361B40C-C042-40E4-8CD1-5A24EA120E9B}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/picomodellib.vcproj
===================================================================
--- tools/vcprojects/picomodellib.vcproj	(revision 3696)
+++ tools/vcprojects/picomodellib.vcproj	(working copy)
@@ -1,10 +1,11 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="picomodellib"
 	ProjectGUID="{8050CFB1-20CC-40FD-8D74-0901C9CA2BE8}"
 	RootNamespace="picomodellib"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
Index: tools/vcprojects/scenegraph.vcproj
===================================================================
--- tools/vcprojects/scenegraph.vcproj	(revision 3696)
+++ tools/vcprojects/scenegraph.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="scenegraph"
 	ProjectGUID="{89043465-96F4-4E90-A351-18ECEBD7E6F2}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -81,6 +82,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -103,9 +106,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -177,9 +177,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -200,9 +202,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/scenelib.vcproj
===================================================================
--- tools/vcprojects/scenelib.vcproj	(revision 3696)
+++ tools/vcprojects/scenelib.vcproj	(working copy)
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="scenelib"
 	ProjectGUID="{F7408B46-E4A9-470C-9731-9A1564247385}"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
Index: tools/vcprojects/shaders.vcproj
===================================================================
--- tools/vcprojects/shaders.vcproj	(revision 3696)
+++ tools/vcprojects/shaders.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="shaders"
 	ProjectGUID="{D93EE174-6724-4F9A-B58D-11EE7EC92674}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/skins.vcproj
===================================================================
--- tools/vcprojects/skins.vcproj	(revision 3696)
+++ tools/vcprojects/skins.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="skins"
 	ProjectGUID="{CF4FE910-F038-4E19-A1E1-E11B584873F0}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/sound.vcproj
===================================================================
--- tools/vcprojects/sound.vcproj	(revision 3696)
+++ tools/vcprojects/sound.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="sound"
 	ProjectGUID="{1482AAB4-40B6-44C0-9FD8-AF80C18A5E7F}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -83,6 +84,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -105,9 +108,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -181,9 +181,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -204,9 +206,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/uimanager.vcproj
===================================================================
--- tools/vcprojects/uimanager.vcproj	(revision 3696)
+++ tools/vcprojects/uimanager.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="uimanager"
 	ProjectGUID="{97E454CC-E8B0-4274-AEAE-5157F420C984}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/undo.vcproj
===================================================================
--- tools/vcprojects/undo.vcproj	(revision 3696)
+++ tools/vcprojects/undo.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="undo"
 	ProjectGUID="{4BCE007A-3FD5-404B-9624-727736437EB0}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/vfspk3.vcproj
===================================================================
--- tools/vcprojects/vfspk3.vcproj	(revision 3696)
+++ tools/vcprojects/vfspk3.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="vfspk3"
 	ProjectGUID="{D933EF8B-2C21-46E7-B720-0621C090AE6E}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/xmlregistry.vcproj
===================================================================
--- tools/vcprojects/xmlregistry.vcproj	(revision 3696)
+++ tools/vcprojects/xmlregistry.vcproj	(working copy)
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="xmlregistry"
 	ProjectGUID="{AC08880C-1150-4579-8870-A8C47C8E5B61}"
 	RootNamespace="archivezip"
 	Keyword="Win32Proj"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
@@ -82,6 +83,8 @@
 				GenerateDebugInformation="true"
 				ProgramDatabaseFile="$(OutDir)/$(ProjectName).pdb"
 				SubSystem="2"
+				RandomizedBaseAddress="1"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
 			/>
@@ -104,9 +107,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
@@ -179,9 +179,11 @@
 				SubSystem="2"
 				OptimizeReferences="2"
 				EnableCOMDATFolding="2"
+				RandomizedBaseAddress="1"
+				FixedBaseAddress="0"
+				DataExecutionPrevention="0"
 				ImportLibrary=""
 				TargetMachine="1"
-				FixedBaseAddress="0"
 			/>
 			<Tool
 				Name="VCALinkTool"
@@ -202,9 +204,6 @@
 				Name="VCAppVerifierTool"
 			/>
 			<Tool
-				Name="VCWebDeploymentTool"
-			/>
-			<Tool
 				Name="VCPostBuildEventTool"
 				CommandLine=""
 			/>
Index: tools/vcprojects/xmlutillib.vcproj
===================================================================
--- tools/vcprojects/xmlutillib.vcproj	(revision 3696)
+++ tools/vcprojects/xmlutillib.vcproj	(working copy)
@@ -1,10 +1,11 @@
 <?xml version="1.0" encoding="Windows-1252"?>
 <VisualStudioProject
 	ProjectType="Visual C++"
-	Version="8,00"
+	Version="9.00"
 	Name="xmlutillib"
 	ProjectGUID="{A15EFB56-927F-411D-A57B-0328321456A2}"
 	RootNamespace="xmlutillib"
+	TargetFrameworkVersion="131072"
 	>
 	<Platforms>
 		<Platform
